### ISSUE TITLE
Issue with self.has_exception() returning False in a failing test case.

### ISSUE THREAD

**Issue by coseto6125** [seleniumbase/SeleniumBase#1861]
I ran a failing test case and noticed that when the tearDown method was executed, 
self.has_exception() returned False. Does this behavior indicate that there is an issue? 

What settings should I configure to ensure that it can accurately detect exceptions in failing test cases?

the code copy from  "BaseCase subclass inheritance"  and  example "test_fail.py" , 
and I ran in terminal cmd "pytest test_fail.py --dashboard --html=report.html"

```python
from seleniumbase import BaseCase

BaseCase.main(__name__, __file__)


class BaseTestCase(BaseCase):

    def setUp(self):
        super().setUp()
        # <<< Run custom setUp() code for tests AFTER the super().setUp() >>>

    def tearDown(self):
        self.save_teardown_screenshot()  # If test fails, or if "--screenshot"
        if self.has_exception():
            # <<< Run custom code if the test failed. >>>
            print("fail")
        else:
            # <<< Run custom code if the test passed. >>>
            print("passed")
        # (Wrap unreliable tearDown() code in a try/except block.)
        # <<< Run custom tearDown() code BEFORE the super().tearDown() >>>
        super().tearDown()


class TestFailing(BaseTestCase):

    def test_find_army_of_robots_on_xkcd_desert_island(self):
        self.open("https://xkcd.com/731/")
        print("\n(This test should fail)")
        self.assert_element("div#ARMY_OF_ROBOTS", timeout=1)
```
<img width="1675" alt="image" src="https://user-images.githubusercontent.com/80243681/235362886-f209ae68-f4d2-4689-8f33-dad126e72ba8.png">

**Comment by mdmintz** [comment]
Looks like an issue starting with Python 3.11. This is working correctly on Python 3.10 and below. Thanks for reporting!

**Comment by mdmintz** [comment]
This was resolved in ``4.14.5`` - https://github.com/seleniumbase/SeleniumBase/releases/tag/v4.14.5
